3-reveal/migrations/5-IRS/2-Zambia-2019/deploy/process_structure_geo_hierarchy_queue.psql

```diff
@@ -8,10 +8,52 @@ BEGIN;
 -- :"schema" variable - and default to public.
 SET search_path TO :"schema",public;
 
+CREATE FUNCTION process_structure_geo_hierarchy_full() RETURNS VOID AS $$
+-- this function is used when we need to reprocess all locations
+-- WARNING: this is an expensive query!!
+BEGIN
+    -- first drop structure_geo_hierarchy_full
+    DROP MATERIALIZED VIEW IF EXISTS structure_geo_hierarchy_full;
+    -- then recreate it
+    CREATE MATERIALIZED VIEW structure_geo_hierarchy_full AS
+    SELECT
+            DISTINCT ON (locations.id) locations.id AS id,
+            jurisdictions.id AS jurisdiction_id,
+            jurisdictions.geographic_level AS jurisdiction_depth,
+            geo_jurisdictions.id AS geo_jurisdiction_id,
+            geo_jurisdictions.geographic_level AS geo_jurisdiction_depth,
+            public.ST_WITHIN(locations.geometry, geo_jurisdictions.geometry) AS geo_strict_within
+        FROM locations
+        LEFT JOIN jurisdictions
+            AS geo_jurisdictions
+            ON public.ST_DWITHIN(locations.geometry, geo_jurisdictions.geometry, 150.0 / 111111.0)
+        LEFT JOIN jurisdictions
+            AS jurisdictions
+            ON jurisdictions.id = locations.jurisdiction_id
```

Comment:
is this indexed?

Reply:
Yes, they are indexed [here (id)](https://github.com/OpenSRP/opensrp-reveal-datawarehouse/blob/master/3-reveal/migrations/2-transaction_tables/deploy/jurisdictions.psql#L23) and [here (jurisdiction_id)](https://github.com/OpenSRP/opensrp-reveal-datawarehouse/blob/master/3-reveal/migrations/2-transaction_tables/deploy/locations.psql#L39)